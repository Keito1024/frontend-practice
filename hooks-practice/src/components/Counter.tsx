import React, { useReducer } from 'react';

interface CounterState {
  count: number
}

interface CounterAction {
  type: ActionType,
  payload: CounterState
}

enum ActionType {
  INCREMENT = 'increment',
  DECREMENT = 'decrement'
}

const initialState = { count: 0 };

const reducer: React.Reducer<CounterState, CounterAction> = (state: CounterState, action: CounterAction) => {
  switch (action.type) {
    case 'increment':
      return {count: state.count + 1};
    case 'decrement':
      return {count: state.count - 1};
    default:
      throw new Error();
  }
}

export const Counter: React.FC = () => {
  const [state, dispatch] = useReducer(reducer, initialState)
  return (
    <>
      Count: {state.count}
      <button onClick={() => dispatch({type: ActionType.DECREMENT, payload: {...state}})}>-</button>
      <button onClick={() => dispatch({type: ActionType.INCREMENT, payload: {...state}})}>+</button>
    </>
  );
}

export default Counter;