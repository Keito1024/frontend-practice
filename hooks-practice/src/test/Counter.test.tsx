import React from 'react';
import renderer from 'react-test-renderer';

import Counter from '../components/Counter';

test('render test', () => {
  const component = renderer.create(
    <Counter />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})