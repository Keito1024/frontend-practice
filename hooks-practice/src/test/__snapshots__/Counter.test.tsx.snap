// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render test 1`] = `
<div>
  Count: 
  0
  <button
    onClick={[Function]}
  >
    -
  </button>
  <button
    className="css-3mgw0s-Counter"
    onClick={[Function]}
  >
    +
  </button>
</div>
`;
